src/main/java/com/cvsgo/repository/ProductCustomRepositoryImpl.java

```diff
@@ -86,6 +89,23 @@ public Page<ProductResponseDto> searchByFilter(User user, ProductSearchFilter fi
         return new PageImpl<>(results, pageable, totalCount);
     }
 
+    public Optional<ProductDetailResponseDto> findByProductId(User user, Long productId) {
+        return Optional.ofNullable(queryFactory
+            .select(new QProductDetailResponseDto(
+                product.as("product"),
+                product.manufacturer.as("manufacturer"),
+                ExpressionUtils.as(JPAExpressions.selectOne().from(productLike)
+                    .where(product.id.eq(productLike.product.id)
+                        .and(findUserProductLike(user))).limit(1), "isLiked"),
+                ExpressionUtils.as(JPAExpressions.selectOne().from(productBookmark)
+                    .where(product.id.eq(productBookmark.product.id)
+                        .and(findUserProductBookmark(user))).limit(1), "isBookmarked")
+            ))
+            .from(product)
+            .where(product.id.eq(productId))
```

Comment:
서브쿼리는 성능상 좋지 않다고 합니다! https://jojoldu.tistory.com/379
left outer join을 통해 ProductBookmark, ProductLike를 조회하도록 수정해보았는데요, (해당 값이 null이면 좋아요/북마크를 하지 않은 것)
이렇게 하면 어떤 필드들을 select하는지 한눈에 보여서 가독성 측면에서도 더 좋을 것 같은데 이 방법은 어떤가요?
```suggestion
            .select(new QProductDetailResponseDto(
                product.as("product"),
                product.manufacturer.as("manufacturer"),
                productLike.as("productLike"),
                productBookmark.as("productBookmark")
            ))
            .from(product)
            .leftJoin(productLike)
                .on(productLike.product.id.eq(productId).and(productLike.user.eq(user)))
            .leftJoin(productBookmark)
                .on(productBookmark.product.id.eq(productId).and(productBookmark.user.eq(user)))
            .where(product.id.eq(productId))
```

Reply:
정말 좋은것같아요 👍👍👍